src/Components/Server/src/Circuits/RemoteJSDataStream.cs

```diff
@@ -181,28 +181,14 @@ public override void Write(byte[] buffer, int offset, int count)
 
     public override async Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
     {
-        var linkedCancellationToken = GetLinkedCancellationToken(_streamCancellationToken, cancellationToken);
-        return await _pipeReaderStream.ReadAsync(buffer.AsMemory(offset, count), linkedCancellationToken);
+        using var linkedCts = ValueLinkedCancellationTokenSource.Create(_streamCancellationToken, cancellationToken);
+        return await _pipeReaderStream.ReadAsync(buffer.AsMemory(offset, count), linkedCts.Token);
     }
 
     public override async ValueTask<int> ReadAsync(Memory<byte> buffer, CancellationToken cancellationToken = default)
     {
-        var linkedCancellationToken = GetLinkedCancellationToken(_streamCancellationToken, cancellationToken);
-        return await _pipeReaderStream.ReadAsync(buffer, linkedCancellationToken);
-    }
-
-    private static CancellationToken GetLinkedCancellationToken(CancellationToken a, CancellationToken b)
-    {
-        if (a.CanBeCanceled && b.CanBeCanceled)
-        {
-            return CancellationTokenSource.CreateLinkedTokenSource(a, b).Token;
-        }
-        else if (a.CanBeCanceled)
-        {
-            return a;
-        }
```

Comment:
What's the reason for this logic disappearing and relying directly on `ValueLinkdedCancellationTokenSource.Create`

Reply:
Nevermind, I saw it got moved. I thought `ValueLinkdedCancellationTokenSource` was a framework class.